The original patch was from an ArchLinux package
https://github.com/archlinux/svntogit-community/blob/packages/vokoscreen/trunk/0001-Add-better-integration-for-Linux.patch

I got this one from https://github.com/City-busz/vokoscreenNG branch 'linux-support', and modified it to work with 3.1.0

commit 0a378409 (HEAD -> linux-support, origin/linux-support)
Author: Balló György <ballogyor@gmail.com>
Date:   2020-07-06 07:14:26 -0600

    Add better integration for Linux
    
    Add AppStream metadata, add installation target.

---


diff --git a/src/applications/INFO b/src/applications/INFO
index 4195de8..1c1bd87 100644
--- a/src/applications/INFO
+++ b/src/applications/INFO
@@ -1,2 +1,3 @@
+vokoscreenNG.appdata.xml
 vokoscreenNG.desktop
 vokoscreenNG.png
diff --git a/src/applications/vokoscreenNG.appdata.xml b/src/applications/vokoscreenNG.appdata.xml
new file mode 100644
index 0000000..883f4ae
--- /dev/null
+++ b/src/applications/vokoscreenNG.appdata.xml
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>vokoscreenNG.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0</project_license>
+  <name>vokoscreenNG</name>
+  <summary>Easy to use desktop recorder</summary>
+  <description>
+    <p>vokoscreenNG is an easy to use screencast creator to record educational videos,
+       live recordings of browser, installation, videoconferences, etc.</p>
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-0.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-1.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-2.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-3.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-4.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-5.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-6.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-7.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-8.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://vokoscreen.volkoh.de/3.0/picture/vokoscreenNG-9.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://linuxecke.volkoh.de/vokoscreen/vokoscreen.html</url>
+  <url type="bugtracker">https://github.com/vkohaupt/vokoscreenNG/issues</url>
+  <url type="donation">https://linuxecke.volkoh.de/vokoscreen/vokoscreen-donate.html</url>
+  <developer_name>Volker Kohaupt</developer_name>
+</component>
--- a/src/vokoscreenNG.pro	2025-03-24 02:28:38.000000000 -0600
+++ b/src/vokoscreenNG.pro	2025-06-24 21:29:38.433946556 -0600
@@ -220,3 +220,28 @@

 # spezialProgressBarAudio
 include(spezialProgressBarAudio/spezialProgressBarAudio.pri)
+
+unix:!macx {
+  isEmpty(PREFIX) {
+    PREFIX = /usr/local
+  }
+  isEmpty(BINDIR) {
+    BINDIR = $$PREFIX/bin
+  }
+  isEmpty(DATADIR) {
+    DATADIR = $$PREFIX/share
+  }
+
+  target.path = $$BINDIR
+
+  icon.files = applications/vokoscreenNG.png
+  icon.path = $$DATADIR/icons/hicolor/256x256/apps/
+
+  desktop.files = applications/vokoscreenNG.desktop
+  desktop.path = $$DATADIR/applications/
+
+  appdata.files = applications/vokoscreenNG.appdata.xml
+  appdata.path = $$DATADIR/metainfo/
+
+  INSTALLS += target icon desktop appdata
+}
